File: leetcode/3371.Identify_the_Largest_Outlier_in_an_Array.cpp
class Solution {
public:
    int getLargestOutlier(vector<int>& nums) {
        long int sum = 0;
        map<int, int> umap;
        for (int num: nums) {
            sum += static_cast<long int>(num);
            if (umap.find(num) == umap.end()) {umap[num] = 1;}
            else {umap[num] += 1;}
        }
        int largest_outlier = INT_MIN;
        for (int num: nums) {
            long int temp = sum - num;
            if ((temp % 2 != 0) || (umap.find(temp/2) == umap.end())) {continue;}
            if (num == temp/2 && umap[num] == 1) {continue;}
            if (num > largest_outlier) {largest_outlier = num;}
        }
        return largest_outlier;
    }
};

File: leetcode/3371.Identify_the_Largest_Outlier_in_an_Array.java
class Solution {
    public int getLargestOutlier(int[] nums) {
        long sum = 0;
        HashMap<Integer, Integer> umap = new HashMap<>();
        for (int num: nums) {
            sum += num;
            umap.put(num, umap.getOrDefault(num, 0) + 1);
        }
        int largest_outlier = Integer.MIN_VALUE;
        for (int num: nums) {
            long temp = sum - num;
            if ((temp % 2 != 0) || (!umap.containsKey((int)temp/2))) {continue;}
            if (num == temp/2 && umap.get(num) == 1) {continue;}
            if (num > largest_outlier) {largest_outlier = num;}
        }
        return largest_outlier;
    }
};
